pkg/auth/providers/azure/clients/adclient/ad_graph_client.go

```diff
@@ -1,28 +1,58 @@
-package azure
+package adclient
 
 import (
-	"encoding/base64"
-	"encoding/json"
+	"context"
 	"strings"
 
-	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
-
 	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
-	"github.com/rancher/norman/httperror"
-	"github.com/sirupsen/logrus"
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/providers/azure/clients"
 )
 
-type azureClient struct {
+type azureADGraphClient struct {
 	servicePrincipal *adal.ServicePrincipalToken
 	userClient       graphrbac.UsersClient
 	groupClient      graphrbac.GroupsClient
 }
 
-// newClientCode sets up the SPT, user and group client using a code
-func newClientCode(code string, config *v32.AzureADConfig) (*azureClient, error) {
-	ac := &azureClient{}
+func (c azureADGraphClient) GetUser(id string) (interface{}, error) {
+	return c.userClient.Get(context.Background(), id)
+}
+
+func (c azureADGraphClient) ListUsers(filter string) (interface{}, error) {
+	return c.userClient.List(context.Background(), filter, "")
+}
+
+func (c azureADGraphClient) GetGroup(id string) (interface{}, error) {
+	return c.groupClient.Get(context.Background(), id)
+}
+
+func (c azureADGraphClient) ListGroups(filter string) (interface{}, error) {
+	return c.groupClient.List(context.Background(), filter)
+}
+
+func (c azureADGraphClient) ListGroupMemberships(id string, params interface{}) ([]string, error) {
+	p, ok := params.(graphrbac.UserGetMemberGroupsParameters)
```

Comment:
It looks like the old client is using one parameter - `SecurityEnabledOnly`. You could probably skip providing this as an arg and hardcode it in this function. 

This would have the following benefits:
1) Puts the logic related to the old api in the client specific to the old api (rather than having an old api filter in the more generic provider).
2) Prevents you from having to do a type cast here.
3) Reduces the number of required args to call this function.